src/mirth/elab.mth

```diff
@@ -981,14 +981,14 @@ def(data-word-new!, Data Str Int -- Word,
 
 def(elab-data-done!, Data --,
     \(dat ->
-        dat is-enum? then(
-            dat "tag" 0 data-word-new! \(tag ->
-                CTX0 dat TData T1 TYPE_INT T1 T-> ready2 tag ~ctx-type !
-                tag ab-build-word!(
-                    COERCE_UNSAFE ab-coerce!
-                ) drop
-            )
+        dat "tag" 0 data-word-new! \(tag ->
+            CTX0 dat TData T1 TYPE_INT T1 T-> ready2 tag ~ctx-type !
+            tag ab-build-word!(
+                COERCE_UNSAFE ab-coerce!
```

Comment:
`COERCE_UNSAFE` won't work for non-enums! It's only does the right thing if the value doesn't have a payload. Because otherwise the underlying value is a tuple, and "coerce" will not get the actual tag that's stored in the tuple.

To make this work for non-enums, we need a new kind of `op`, say `OP_GET_TAG -> TData`, which calls `get_data_tag(v)` in the C99 runtime.

Reply:
oh dang, you're right, all the places I was using it were non tuple variants so that explains why it didn't seem to break anything

Reply:
Oh one more thing I just remembered: it's probably a good idea to only generate `tag` for non-resource types.

Reply:
yep, probably good to do that, that one is very easy to fix too.